ad-server/src/queue.rs

```diff
@@ -0,0 +1,217 @@
+use std::{collections::HashSet, sync::Arc};
+
+use alloy::primitives::FixedBytes;
+use anyhow::Result;
+use app::{DEPTH, Helper};
+use common::{
+    circuits::shrink_compress_pod,
+    payload::{Payload, PayloadInit, PayloadUpdate},
+};
+use pod2::{
+    backends::plonky2::mainpod::Prover,
+    dict,
+    frontend::MainPodBuilder,
+    middleware::{Hash, RawValue, Value, containers},
+};
+use serde::{Deserialize, Serialize};
+use tokio::{sync::mpsc::Receiver, task};
+use uuid::Uuid;
+
+use crate::{Context, db};
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub enum State {
+    Init(StateInit),
+    Update(StateUpdate),
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub enum StateInit {
+    Pending,
+    SendingBlobTx,
+    Complete { id: i64, tx_hash: FixedBytes<32> },
```

Comment:
is there any reason to not use the `TxType` here? (from `alloy::primitives::TxHash`)
```suggestion
    Complete { id: i64, tx_hash: TxType },
```